src/main/java/gov/va/ascent/framework/rest/provider/RestProviderHttpResponseCodeAspect.java

```diff
@@ -257,19 +263,49 @@ private void writeAudit(Object request, Object response,
         }
 	}
 
-	private void getHttpRequestAuditData(HttpServletRequest httpServletRequest, 
+	private void getHttpRequestAuditData(HttpServletRequest httpServletRequest,
 			RequestResponseAuditData requestResponseAuditData) {
-	    final Map<String, String> headers = new HashMap<>();
+		final Map<String, String> headers = new HashMap<>();
+		Enumeration<String> headerNames = httpServletRequest.getHeaderNames();
 
-        Enumeration<String> headerNames = httpServletRequest.getHeaderNames();
+		while (headerNames.hasMoreElements()) {
+			String headerName = headerNames.nextElement();
+			String value = httpServletRequest.getHeader(headerName);
+			headers.put(headerName, value);
+		}
+		requestResponseAuditData.setHeaders(headers);
+		requestResponseAuditData.setUri(httpServletRequest.getRequestURI());
+		requestResponseAuditData.setMethod(httpServletRequest.getMethod());
+		ArrayList<String> attachmentTextList = new ArrayList<>();
```

Comment:
@rajuthota1 : The type of the "attachmentTextList" object should be an interface such as "List" rather than the implementation "ArrayList"